### PR TITLE
Update metadata proposal with new fields

### PR DESCRIPTION
**Pull request by maelk**
This proposal adds a couple of convenient fields to the CAPM3 metadata proposal

### CODE REVIEW

File: design/metadata-handling.md [hunk 1]
```diff
@@ -421,6 +430,14 @@ ways. The following types of objects are available and accept lists:
...
 - **fromHostInterfaces**: renders the MAC address of the BareMetalHost that
   matches the name given as value.
+- **fromLabels**: renders the content of a label on an object or an empty string
+  if the label is absent. It takes an `object` attribute to specify the type of
+  the object where to fetch the label, and a `label` attribute that contains the
+  label key.
+- **fromAnnotations**: renders the content of a annotation on an object or an
+  empty string if the annotation is absent. It takes an `object` attribute to
+  specify the type of the object where to fetch the annotation, and an
+  `annotation` attribute that contains the annotation key.
```
Comment (dhellmann) [inline design/metadata-handling.md:440]: What are the possible values for the object field? Can we define an enum?

Comment (maelk) [reply design/metadata-handling.md:440]: In the implementation of the current feature, it is an enum. I'll just document it here in the proposal : "machine, metal3machine, baremetalhost" (case insensitive)

**Comment by maelk** [comment]
Let's put a lazy consensus on this one by end of next week, Friday 08th ?

**Pull request merged**